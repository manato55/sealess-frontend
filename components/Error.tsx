import React from 'react'
import {useRouter} from 'next/router'
import { useSetRecoilState } from 'recoil'
import { http } from '../store/atom'

interface Props {
    errorCode: number;
}

export const Error = (props: Props) => {
    const router = useRouter()
    const setHttpStatus = useSetRecoilState(http)


    const back = async() => {
        window.location.href = '/'
    }

    return (
        <>
            {props.errorCode === 404 ?
                <p>not found</p>
            :props.errorCode === 500 ?
                <p>server error</p>
            :''}
            <p onClick={()=>back()}>topへ戻る</p>
        </>
    )
}


export default Error