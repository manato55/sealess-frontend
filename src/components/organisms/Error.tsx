import React from 'react';
import styled from 'styled-components';

interface Props {
  errorCode: number;
}

export const Error = (props: Props): React.ReactElement => {
  return (
    <>
      <ErrorMessageWrapper>
        <SubWrapper>
          {props.errorCode === 404 ? (
            <p>Not Found（お探しのページは見つかりませんでした。）</p>
          ) : props.errorCode === 500 ? (
            <p>Server Error（内部エラーによりページを表示できません。再度お試しください。）</p>
          ) : (
            ''
          )}
          <a href="/">トップページへ戻る</a>
        </SubWrapper>
      </ErrorMessageWrapper>
    </>
  );
};

const ErrorMessageWrapper = styled.div`
  position: relative;
  height: 100vh;
`;

const SubWrapper = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: center;
  width: 100%;

  p {
    font-size: 20px;
    font-weight: bold;
  }

  a {
    color: blue;
    text-decoration: underline;
  }
`;

export default Error;
