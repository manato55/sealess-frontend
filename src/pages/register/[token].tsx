import RegisterToken from '../../components/organisms/RegisterToken';

const Token = () => {
  return (
    <>
      <RegisterToken />
    </>
  );
};

export default Token;
