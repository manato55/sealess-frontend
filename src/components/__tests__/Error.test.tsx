import React from 'react';
import Error from '../organisms/Error';
import 'jsdom-global/register';
import renderer from 'react-test-renderer';

describe('error test', () => {
  test('snapshot 404', () => {
    const tree = renderer.create(<Error errorCode={404} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('snapshot 500', () => {
    const tree = renderer.create(<Error errorCode={500} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
