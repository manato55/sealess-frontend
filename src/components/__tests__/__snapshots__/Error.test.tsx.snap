// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error test snapshot 404 1`] = `
Array [
  <p>
    not found
  </p>,
  <p
    onClick={[Function]}
  >
    topへ戻る
  </p>,
]
`;

exports[`error test snapshot 500 1`] = `
Array [
  <p>
    server error
  </p>,
  <p
    onClick={[Function]}
  >
    topへ戻る
  </p>,
]
`;
